xdsl/dialects/math.py

```diff
@@ -53,39 +53,23 @@ def __init__(self, operand: Operation | SSAValue):
 
 
 class FloatingPointLikeUnaryMathOperation(IRDLOperation, abc.ABC):
-    """A generic floating-point-like unary math operation."""
+    """A generic floating-point-like unary math operation with fastmath flags."""
 
     T: ClassVar = VarConstraint("T", floatingPointLike)
 
     operand = operand_def(T)
     result = result_def(T)
 
-    assembly_format = "$operand attr-dict `:` type($result)"
-
-    def __init__(self, operand: Operation | SSAValue):
-        operand = SSAValue.get(operand)
-        super().__init__(
-            operands=[operand],
-            result_types=[operand.type],
-        )
-
-
-class FloatingPointLikeUnaryMathOperationWithFastMath(
-    FloatingPointLikeUnaryMathOperation, abc.ABC
-):
-    """A generic floating-point-like unary math operation with fastmath flags."""
-
-    fastmath = opt_prop_def(FastMathFlagsAttr)
+    fastmath = opt_prop_def(FastMathFlagsAttr, default_value=FastMathFlagsAttr("none"))
```

Comment:
Feels like it should not be optional, as it has a default value, which comes pretty close to being the same thing. Does this work?
```suggestion
    fastmath = prop_def(FastMathFlagsAttr, default_value=FastMathFlagsAttr("none"))
```

Reply:
Having said that I just looked up the tablegen and these shouldn't be optional